src/providers/sdk/__tests__/Base.test.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint @typescript-eslint/no-explicit-any: 0 */ // --> OFF
+import { CryptKeeperInjectedProvider } from "..";
+
+jest.mock("@src/background/services/event", (): unknown => ({
+  __esModule: true,
+  default: jest.fn().mockImplementation(),
+}));
+
+jest.mock("@src/background/services/zkProof", (): unknown => ({
+  __esModule: true,
+  default: {
+    getInstance: jest.fn(),
+  },
+}));
+
+describe("providers/sdk/Base", () => {
+  const mockApprovals = {
+    isApproved: true,
+    canSkipApprove: true,
+  };
+
+  beforeEach(() => {
+    // Mocking private dependencies
+    const mockTryInject = jest.spyOn(CryptKeeperInjectedProvider.prototype as any, "tryInject");
```

Comment:
Private dependencies is not the same as private methods.
Here you need to spy on `window.postMessage` and check that it has been called with specific params